File: HappyPuppy/C#/AppEnv.cs

public class AppEnv
{
    public static HappyPuppyRepository repo = null;

    public static AdoptPuppyRequest AdoptPuppy
    {
        get { return new AdoptPuppyUseCase(repo); }
    }

    public static ActionRequest ListPuppies
    {
        get { return new ListPuppiesUseCase(repo); }
    }

    public static ActionRequest ListOwners
    {
        get { return new ListOwnersUseCase(repo); }
    }
}

File: HappyPuppy/Java/AppEnv.java

public class AppEnv
{
    public static Repository.HappyPuppyRepository repo = null;

    public static Boundaries.AdoptPuppyRequest AdoptPuppy()
    { 
        return new UseCases.AdoptPuppyUseCase(repo);
    }

    public static Boundaries.ActionRequest ListPuppies()
    {
        return new UseCases.ListPuppiesUseCase(repo); 
    }

    public static Boundaries.ActionRequest ListOwners()
    {
        return new UseCases.ListOwnersUseCase(repo);
    }
}